crates/signal/src/lib.rs

```diff
@@ -65,15 +63,128 @@
 use std::pin::Pin;
```

Comment:
There is a cfg atter above this which prevents this crate from being compiled on windows, we should remove that.